enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,425 @@
+- Release pipeline needs to be setup to construct manifestlist based release payloads.
+- Explicit upgrade support through oc upgrade's `--to-image` option
+- Does not include support for:
+  - image registry - internal image registry will not support manifest listed images.
+  - imagestreams importing full manifestlist content (will continue to import a single image when referencing manifestlists).
+
+Phase 2: Full basic functionality
+- Imagestreams and Imageregistry manifestlist support to import, pull and push manifestlisted images.
+- OLM support for node selection when deploying an operator and ecosystem improvements.
+- Improve console's content filtering by making it architecture aware and passing that information to OLM.
+- oc debug (and other commands) support for compute architecture.
+- support toolbox.
+- support for Power and Z architectures.
+- Hypershift support for heterogeneous management+worker clusters.
+- Upgrade support in CVO/Cincinnati for heterogeneous release payloads through a heterogeneous graph. Migration of a homogeneous cluster
+would be explicitly done through the upgrade command  by specifying the `--to-image` flag.
+
+Phase 3: Quality of life improvements
+- First class install support - ability to create non-control-plane-architecture workers at install time and generate machineconfig
+and machineset manifests for the same.
+- Possibly reduce mirroring costs by looking into things like sparse manifestlist.
+- CVO migration support for a homogeneous cluster to a heterogeneous payload
+
+### User Stories
+
+- As an enterprise customer looking to save compute resource cost, I want to be able to run part of my application workload on ARM instead of x86.
+- As the owner of an application that runs partially on x86 and partially on ARM, I want to be able to run the entire application within a
+single cluster so I do not need to pay the cost of higher latency between clusters, or worry about the HA-resilience of cross-cluster bridges.
+
+### Workflow Description
+
+A heterogeneous cluster installation will have the same experience as installing a homogeneous cluster. This is the long term vision. However in
+the short term, for the 4.11 Tech Preview, some additional steps will be necessary as IPI installs are not supported yet. The flow will be the
+following:
+- Pick a heterogeneous payload to install from https://multi.ocp.releases.ci.openshift.org/
+- override the release image before creating a cluster by setting this env variable - OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
+- install a homogeneous x86 cluster as you normally would on AWS/Azure
+- add machinesets of ARM architecture following [these][creating-machinesets] instructions. Make sure to match the instance types and the
+corresponding bootimages based on the architecture of the instance.
+
+### API Extensions
+
+- Imagestreams API will undergo changes described in this [manifestlist enhancement section][imagestreams-api-changes]
```

Comment:
@dmage i have linked to the manifestlist enhancement which was filed a while ago which details the api changes. is that still relevant or are there additional changes needed ?

Reply:
It is relevant, we expect our implementation to be close to the EP.

Reply:
Yeah i think that's ok. What we want for the purposes of the API review is to be able to look at the changes and ensure that they are compliant with our conventions. Normally we add a golang snippet for small changes or link to an openshift/api PR for larger changes. If you have the go code in the manifest list enhancement we can review it there

Reply:
yes it is there- https://github.com/openshift/enhancements/blob/master/enhancements/manifestlist/manifestlist-support.mdit was filed a while ago and was merged. i've linked to this enhancement here